src/board.js

```diff
@@ -98,6 +99,64 @@ class Board {
       }
     }
   }
+
+  detectWhenKingIsInCheck() {    
+    let whiteKingPosition, blackKingPosition;
+
+    // Znalezienie pozycji królów
+    this.forEachSquare((row, column) => {
```

Comment:
Tak sobie teraz myślę że dużo by uprościło gdybyśmy nie musieli polegać na iteracji po `this.squares` by znaleźć bierkę. Można by przy ustawianiu bierek oprócz przypisywanie bierek do pól dodatkowo przypisać bierke do jakieś struktury danych (array albo lepiej może nawet Map) która pozwoli nam wygodniej znajdywać bierki.

Alternatywnie można by było wyciągnąć jakąś metode która wyciągnie nam wszystkie bierki na podstawie `this.square`